libImaging/Effects.c

```diff
@@ -121,14 +121,14 @@ ImagingEffectSpread(Imaging imIn, int distance)
     Imaging imOut;
     int x, y;
 
-    imOut = ImagingNew(imIn->mode, imIn->xsize, imIn->ysize);
+    imOut = ImagingNewDirty(imIn->mode, imIn->xsize, imIn->ysize);
 
     if (!imOut)
-    return NULL;
+        return NULL;
 
 #define SPREAD(type, image)\
-    for (y = 0; y < imIn->ysize; y++)\
-    for (x = 0; x < imIn->xsize; x++) {\
+    for (y = 0; y < imOut->ysize; y++)\
+        for (x = 0; x < imOut->xsize; x++) {\
```

Comment:
In this case `imIn->xsize` == `imOut->xsize` and `imIn->ysize` = `imOut->ysize`. imOut more explicit points to actual action.